mathics/builtin/combinatorial.py

```diff
@@ -370,32 +373,27 @@ def apply(self, list, evaluation):
     def apply_1(self, list, n, evaluation):
         'Subsets[list_?ListQ, n_]'
         
-        tlist = [x for x in list.leaves]
         expr = Expression('Subsets', list, n)
         
-        if not n.get_head_name() == 'System`Integer' or n.to_python() < 0 :
+        n_value = n.get_int_value() 
+        if n_value == 0:
+            return Expression('List', Expression('List'))
+        if n_value is None or n_value < 0:
```

Comment:
I think the order of these if statements should be swapped. Can you add a test, e.g. `Subsets[{1, 2}, x]`